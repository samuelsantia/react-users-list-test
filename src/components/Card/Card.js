import styled from 'styled-components/macro';

/** styled card */
const Card = styled.article`
  display: flex;
  flex-direction: var(--card-direction, row);
  background-color: var(--card-background-color, white);
  border-radius: var(--card-border-radius, 5px);
  box-shadow: var(--card-shadow, none);
  overflow: hidden;
`;

export default Card;
