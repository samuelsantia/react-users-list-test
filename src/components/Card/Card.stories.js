import React from 'react';
import styled from 'styled-components/macro';

import Card from './index';
import CardContent from '../CardContent';
import Typography from '../Typography';

const StoryDecorator = styled.div`
  padding: var(--scale-rhythm);
`;

export default {
  title: 'Card',
  component: Card,
  subcomponents: { CardContent, Typography },
  decorators: [(storyFn) => <StoryDecorator>{storyFn()}</StoryDecorator>],
};

export const basic = () => (
  <Card>
    <CardContent>
      <Typography variant="h5" tag="h2" trailer="1" weight="regular">
        Title
      </Typography>
      <Typography variant="body2">
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam auctor
        suscipit nisl at laoreet. Quisque nisi dolor, suscipit sed laoreet et,
        vestibulum eu nunc. Phasellus imperdiet eros eget eros lacinia iaculis.
        Fusce vehicula diam est, venenatis varius mauris rutrum ut. Duis quis
        mattis ex, dictum dignissim elit. Pellentesque sit amet faucibus arcu.
        Sed vehicula leo ut turpis venenatis convallis. Nulla pellentesque nisi
        turpis, nec posuere massa condimentum non. Donec dolor ipsum, mattis at
        mi vel, vestibulum mattis mauris.
      </Typography>
    </CardContent>
  </Card>
);
