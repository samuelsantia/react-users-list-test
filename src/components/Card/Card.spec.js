import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import Card from './Card';

describe('<Card />', () => {
  it('should render', () => {
    const { container } = render(<Card>Test Card</Card>);

    expect(container.firstChild).toMatchSnapshot();
  });
});
