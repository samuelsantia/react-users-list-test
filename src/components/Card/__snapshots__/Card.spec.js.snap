// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: var(--card-direction,row);
  -ms-flex-direction: var(--card-direction,row);
  flex-direction: var(--card-direction,row);
  background-color: var(--card-background-color,white);
  border-radius: var(--card-border-radius,5px);
  box-shadow: var(--card-shadow,none);
  overflow: hidden;
}

<article
  class="c0"
>
  Test Card
</article>
`;
